### PR TITLE
ENH: Add option to only merge column header cells in `ExcelFormatter`.

### LINKED ISSUE #35384: ENH: The parameter merge_cells parameter in function to_excel should take a 2-tuple one for index and one for columns
Currently pandas `to_excel` function has one parameter named `merge_cells`: if `True` it merges all cells in the excel sheet corresponding to the index and columns. However, if there are many levels of the index and many values for the higher levels, then as we scroll on the sheet, values of the higher levels in index become invisible. One way to keep the repeated values is to set `merge_cells = False`. However, in that case the column levels are also affected.

It would be great to have a way to control this aspect of index and columns separately. One way would be to let `merge_cells` to take a 2-tuple of which one value is for the index and the other for the columns.

### PR DESCRIPTION
**Pull request by ulido**
Adds an option to pass `merge_cells="columns"` in `DataFrame.to_excel`. This allows only merging MultiIndex column header cells and prevents index cells to be merged.

- [x] closes #35384
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/formats/excel.py [hunk 1]
```diff
@@ -523,8 +524,11 @@ class ExcelFormatter:
         Column label for index column(s) if desired. If None is given, and
         `header` and `index` are True, then the index names are used. A
         sequence should be given if the DataFrame uses MultiIndex.
-    merge_cells : bool, default False
-        Format MultiIndex and Hierarchical Rows as merged cells.
+    merge_cells : bool or 'columns', default False
+        Format MultiIndex column headers and Hierarchical Rows as mergedcells
+        if True. Merge MultiIndex column headers only if 'columns'.
+        .. versionchanged:: :ref: 3.0.0
```
Comment (mroeschke) [inline pandas/io/formats/excel.py:530]:
```suggestion
        .. versionchanged:: 3.0.0
```

File: pandas/io/formats/excel.py [hunk 2]
```diff
@@ -523,8 +524,11 @@ class ExcelFormatter:
         Column label for index column(s) if desired. If None is given, and
         `header` and `index` are True, then the index names are used. A
         sequence should be given if the DataFrame uses MultiIndex.
-    merge_cells : bool, default False
-        Format MultiIndex and Hierarchical Rows as merged cells.
+    merge_cells : bool or 'columns', default False
+        Format MultiIndex column headers and Hierarchical Rows as mergedcells
```
Comment (mroeschke) [inline pandas/io/formats/excel.py:528]:
```suggestion
        Format MultiIndex column headers and Hierarchical Rows as merged cells
```

File: pandas/io/formats/excel.py [hunk 3]
```diff
@@ -580,6 +584,9 @@ def __init__(
         self.index = index
         self.index_label = index_label
         self.header = header
+
+        if (not isinstance(merge_cells, bool)) and (merge_cells != "columns"):
+            raise ValueError("Unexpected value for 'merge_cells'.")
```
Comment (mroeschke) [inline pandas/io/formats/excel.py:589]:
```suggestion
        if not isinstance(merge_cells, bool) and merge_cells != "columns":
            raise ValueError(f"Unexpected value for {merge_cells=}.)
```

**Comment by ulido** [comment]
Thanks for the review @mroeschke !

File: doc/source/whatsnew/v3.0.0.rst [hunk 4]
```diff
@@ -45,6 +45,7 @@ Other enhancements
 - :meth:`DataFrame.pivot_table` and :func:`pivot_table` now allow the passing of keyword arguments to ``aggfunc`` through ``**kwargs`` (:issue:`57884`)
 - :meth:`Series.cummin` and :meth:`Series.cummax` now supports :class:`CategoricalDtype` (:issue:`52335`)
 - :meth:`Series.plot` now correctly handle the ``ylabel`` parameter for pie charts, allowing for explicit control over the y-axis label (:issue:`58239`)
+- Add option to only merge MultiIndex column header header cells in :func:`DataFrame.to_excel` (:issue:`35384`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:48]:
```suggestion
- :func:`DataFrame.to_excel` argument ``merge_cells`` now accepts a value of ``"columns"`` to only merge :class:`MultiIndex` column header header cells (:issue:`35384`)
```

**Comment by mroeschke** [comment]
pre-commit.ci autofix

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ulido

**Pull request merged**